examples/aws-cloudwatch/README.md

```diff
@@ -0,0 +1,181 @@
+
+- mysql: A database to hold historical aws information.
+- mysql-createdb: Runs [create_aws_db.sql](./create_aws_db.sql), a script to create the database and table used to hold historical aws information.
+- juttle-aws-poller: Runs [poll_aws.juttle](./poll_aws.juttle), a juttle program that periodically polls from aws, aggregates the raw points into aggregate and demographic information, and writes that information to the mysql database.
+
+## ``juttle-config.json`` configuration
+
+Modify `juttle-config.json` to add ``aws`` and ``cloudwatch`` sections containing credentials to access messages via the AWS and Cloudwatch APIs. As these examples use two separate adapters, both need to be configured.
+
+{
+  "adapters": {
+    "aws": {
+      "access_key": "--YOUR-AWS-ACCESS-KEY-HERE--",
+      "secret_key": "--YOUR-AWS-SECRET-KEY-HERE--",
+      "region": "--YOUR-REGION--HERE"
+    },
+    "cloudwatch": {
+      "access_key": "--YOUR-AWS-ACCESS-KEY-HERE--",
+      "secret_key": "--YOUR-AWS-SECRET-KEY-HERE--",
+      "region": "--YOUR-REGION--HERE"
+    },
+  }
+}
+
+The full set of steps to generate these credentials is [on the README](https://github.com/juttle/juttle-aws-adapter) page on github.
+
+## Juttle Programs
+
+To run any of these programs, just visit
+``http://(localhost|docker machine ip):8080/?path=/examples/aws-cloudwatch/index.juttle``
+and follow the links.
+
+### Polling script and `aggregate_all` subgraph
+
+The polling script [poll_aws.juttle](./poll_aws.juttle) uses the `aggregate_all` subgraph implemented in the file [aws_module.juttle](https://github.com/juttle/juttle-aws-adapter/blob/master/aws_module.juttle) to transform the raw points returned from the various AWS APIs into aggregate and demographic summaries of your AWS infrastructure.
+
+It then writes the resulting points to the table `aws_aggregation` in the mysql database. This allows for a source of historical aws information that is persistent across invocations of individual juttle programs.
+
+In turn, when the example programs want to read aws aggregate/demographic information, they read from the mysql database using `read mysql -table 'aws_aggregation' ...`.
+
+### AWS Overview
+
+This program demonstrates ways to use the aggregate and demographic information from the aws adapter as well as ways to use the cloudwatch adapter to identify problems.
+
+The program shows a sample of aggregate and demographic information from AWS:
+
+- The number of EC2 instances
+- The total amount of RDS allocated storage
+- The total EBS volume capacity
+
+This program also identifies "problem" items and shows the number of items as a count and details as a table.
+
+For this program, "problem" items are:
+- EC2: Any instances with failed status checks.
+- ELB: Any load balancers where UnHealthyHostCount is > 0
+- Lambda: Any function with errors > 0
+
+View this program: [overview.juttle](./overview.juttle)
```

Comment:
The other READMEs follow this convention--describe the program then provide a link. I filed https://github.com/juttle/juttle-engine/issues/42 for now, if we change it I think we should do it for all READMEs (and in a separate PR).